panther_manager/src/plugins/condition/check_bool_msg.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_manager/plugins/condition/check_bool_msg.hpp"
+
+#include "panther_manager/behavior_tree_utils.hpp"
+
+namespace panther_manager
+{
+
+BT::NodeStatus CheckBoolMsg::onTick(const BoolMsg::SharedPtr & last_msg)
+{
+  bool expected_data;
+  getInput<bool>("data", expected_data);
+
+  return (last_msg && expected_data == last_msg->data) ? BT::NodeStatus::SUCCESS
```

Comment:
I like it as it is. You could add bool is_data_equal = last_msg->data == expected_data, but this is a simple comparison, I don't know if it makes sense. As for logs, the only sensible log-level is debug (we don't want to spam that the result of the check is incorrect, because maybe someone doesn't want to start docking now). At the same time, if necessary, the result can be easily viewed in Groot.

Reply:
Okay